pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -437,6 +436,10 @@ func (r *RootSyncReconciler) SetupWithManager(mgr controllerruntime.Manager, wat
 			handler.EnqueueRequestsFromMapFunc(r.mapObjectToRootSync),
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
 		Watches(&source.Kind{Type: &rbacv1.ClusterRoleBinding{}},
+			handler.EnqueueRequestsFromMapFunc(r.mapObjectToRootSync),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
+		// TODO: is it possible to watch with a label filter?
+		Watches(&source.Kind{Type: &rbacv1.RoleBinding{}},
```

Comment:
Can you only start watches when users opt-in to this feature?

Reply:
My understanding is that the watches are instantiated at controller startup, so I don't think this would be possible given that the feature is enabled on the RootSync.

As a side note I'm looking into adding a label selector for the shared cache, which should optimize all of these watches.

Reply:
Actually, I think we have code already that adds watches at runtime. It's not as pretty as this interface but it's doable. This could be a future optimization.

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/reposync_controller.go#L550